### PR TITLE
Pass opener frame info from UI process to main frame constructor

### PR DESCRIPTION
**Pull request by achristensen07**
#### 4c2316acb3d16926f92634aedb92ecdd0fb32829
<pre>
Pass opener frame info from UI process to main frame constructor
<a href="https://bugs.webkit.org/show_bug.cgi?id=268704">https://bugs.webkit.org/show_bug.cgi?id=268704</a>
<a href="https://rdar.apple.com/122248591">rdar://122248591</a>

Reviewed by Pascoe.

If a page has an opener, with site isolation we will need to get that information
from the UI process if the opener is in a different process than the opened page.

This is the first step to better supporting window.open and iframe combinations
with site isolation.  The next step is having the BrowsingContextGroup not only
keep track of what processes are in the group for what domain, but also keep track
of what pages are in the group so they can all have RemoteFrames injected into a
new process in the group.

* Source/WebCore/loader/EmptyClients.cpp:
(WebCore::pageConfigurationWithEmptyClients):
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::LocalFrame):
(WebCore::LocalFrame::createMainFrame):
(WebCore::LocalFrame::createSubframe):
(WebCore::LocalFrame::createSubframeHostedInAnotherProcess):
* Source/WebCore/page/LocalFrame.h:
* Source/WebCore/page/Page.cpp:
(WebCore::createMainFrame):
(WebCore::Page::Page):
* Source/WebCore/page/PageConfiguration.cpp:
(WebCore::PageConfiguration::PageConfiguration):
* Source/WebCore/page/PageConfiguration.h:
* Source/WebKit/UIProcess/ProvisionalPageProxy.cpp:
(WebKit::ProvisionalPageProxy::initializeWebPage):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::creationParameters):
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::WebFrame::transitionToLocal):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::frameOpener):
(WebKit::m_unifiedTextReplacementController):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _commonInitializationWithFrameName:groupName:]):
(-[WebView initSimpleHTMLDocumentWithStyle:frame:preferences:groupName:]):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/274396@main">https://commits.webkit.org/274396@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/1915104ee5944d2882ba50b5e65ac36bfe76bb63

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/38807 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/17738 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/41152 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/41338 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/34460 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/41114 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/20558 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15087 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/32538 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/39380 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/14925 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/33672 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12967 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/38673 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/12931 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/34573 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/42614 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35223 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/34933 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/38767 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/13617 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11237 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36983 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/15229 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/14895 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5086 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/14710 "Built successfully") | | |

### CODE REVIEW

File: Source/WebKit/WebProcess/WebPage/WebPage.cpp [hunk 1]
```diff
@@ -498,6 +498,16 @@ static std::variant<UniqueRef<LocalFrameLoaderClient>, UniqueRef<RemoteFrameClie
     RELEASE_ASSERT_NOT_REACHED();
 }
 
+static RefPtr<Frame> frameOpener(std::optional<FrameIdentifier> identifier)
```
Comment (pascoej) [inline Source/WebKit/WebProcess/WebPage/WebPage.cpp]: Calling this `frameOpener` feels wrong. It looks up an identifier and returns the Frame for it, if it exists. Calling it `frameFromIdentifier` or something might be better.

**Review by pascoej — APPROVED** [review]
One small comment that I think should be addressed before landing.

File: Source/WebKit/WebProcess/WebPage/WebPage.cpp (hunk 1 above)
Comment (pascoej) [reply Source/WebKit/WebProcess/WebPage/WebPage.cpp]: Making `WebProcess::singleton().webFrame(` take in an optional might be good option too.

Comment (achristensen07) [reply Source/WebKit/WebProcess/WebPage/WebPage.cpp]: frameFromIdentifier is a much better name.  Changing WebProcess::webFrame should probably be done in a separate PR.  It looks like 20 of the 63 call sites would benefit.

**Pull request merged**